### ISSUE TITLE
Hashes are different (

### ISSUE THREAD

**Issue by sokolegg** [calebh/differentiablesha256#1]
Hi, I've tried to reproduce on torch, and torch sha256 differs from real sha256

Is it stable in the last version of pytorch?

**Comment by calebh** [comment]
It's been a long time since I've looked at this code and it's not exactly in the best condition. You might want to try looking at the adder branch instead of master.

The interior of the relaxed space is filled with local minima, so if you're hoping to run gradient descent on it as I was you're out of luck. Even if the surface were to smoothly interpolate between the discrete values of sha256, the problem is still non-convex.

**Comment by sokolegg** [comment]
I have another idea. But your code can be quite helpful for this. 
I want to optimize the computation of sha256 by using a neural network.

@calebh  Could you write me, please?
sokolegg@yandex.ru

I want to discuss this idea with you.
